lightlab/util/data/two_dim.py

```diff
@@ -79,36 +95,106 @@ def __eq__(self, other):
+        ''' This works with scalars, vectors, MeasuredFunctions, and FunctionBundles
+        '''
+        if np.isscalar(other) or isinstance(other, MeasuredFunction):
+            other = repeat(other, self.nDims)
+        newBundle = type(self)()
+        for selfItem, otherItem in zip(self, other):
+            newItem = selfItem + otherItem
+            newBundle.addDim(newItem)
+        return newBundle
 
     def __radd__(self, other):
         return self.__add__(other)
 
+    def __sub__(self, other):
+        return self.__add__(-1 * other)
+
     def __mul__(self, other):
-        if np.isscalar(other):
-            other = np.ones(self.nDims) * other
-        newObj = self.copy()
-        for iDim in range(self.nDims):
-            newObj.ordiMat[iDim] = self.ordiMat[iDim] * other[iDim]
-        return newObj
+        ''' This works with scalars, vectors, MeasuredFunctions, and FunctionBundles
+        '''
+        if np.isscalar(other) or isinstance(other, MeasuredFunction):
+            other = repeat(other, self.nDims)
+        newBundle = type(self)()
+        for selfItem, otherItem in zip(self, other):
+            newItem = selfItem * otherItem
+            newBundle.addDim(newItem)
+        return newBundle
 
     def __rmul__(self, other):
         return self.__mul__(other)
 
     def __truediv__(self, other):
         return self * (1 / other)
 
+    def __getattr__(self, attrName):
+        ''' Feeds through **callable** methods to its members (type MeasuredFunction),
+            if the attribute is not found in the FunctionBundle.
+            If it finds it there, then applies it to every function in the bundle.
+            With binary operator math, it only works with scalars.
+
+            For example, starting with::
+
+            .. code-block:: python
+
+                spct1 = Spectrum([0, 1, 2], [2, 5, 3])
+                spct2 = Spectrum([0, 1, 2], [4, 4, 1])
+                funBun = FunctionBundle([spct1, spct2])
+
+            You can do
+
+            .. code-block:: python
+
+                croppedFunBund = FunctionBundle([
+                                      spct1.crop([0, 1]),
+                                      spct2.crop([0, 1]))
+                                      ])
+                funBun.crop([0, 1]) == croppedFunBund
+
+            Note:
+                Be careful about "overloading" a MeasuredFunction method in FunctionBundle.
+                It can have a different meaning than what would happen if not overloaded.
+                For example ``MeasuredFunction.__mul__(nonscalar)`` is function multiplication,
+                while ``FunctionBundle.__mul__(nonscalar)`` means vector dot product.
+
+            Todo:
+                This only works with :class:`~lightlab.util.data.one_dim.SignalProcessingMixin`
+                methods that are MF-in/MF-out. How should we handle characteristic-type methods,
+                such as ``centerOfMass``, or ``getRange``?
```

Comment:
I'll leave this TODO ignored for now. I think we should just return a list with the results.